frontend/src/components/maps/leaflet/Map.tsx

```diff
@@ -177,52 +194,66 @@ const Map: React.FC<MapProps> = ({
 }) => {
   const keycloak = useKeycloakWrapper();
   const dispatch = useAppDispatch();
-  const [geoFilter, setGeoFilter] = useState<IGeoSearchParams>({
-    ...defaultFilterValues,
-    includeAllProperties: keycloak.hasClaim(Claims.ADMIN_PROPERTIES),
-  } as any);
-  const [baseLayers, setBaseLayers] = useState<BaseLayer[]>([]);
-  const [triggerFilterChanged, setTriggerFilterChanged] = useState(true);
-  const [activeBasemap, setActiveBasemap] = useState<BaseLayer | null>(null);
-  const smallScreen = useMediaQuery({ maxWidth: 1800 });
-  const [mapWidth, setMapWidth] = useState(0);
+  const [triggerFilterChanged, setTriggerFilterChanged] = React.useState(true);
   const municipalitiesService = useLayerQuery(MUNICIPALITY_LAYER_URL);
   const parcelsService = useLayerQuery(PARCELS_PUBLIC_LAYER_URL);
-  const [bounds, setBounds] = useState<LatLngBounds>(defaultBounds);
   const { setChanged } = useFilterContext();
-  const [layerPopup, setLayerPopup] = useState<LayerPopupInformation>();
   const popUpContext = React.useContext(PropertyPopUpContext);
-
-  if (mapRef.current && !selectedProperty?.parcelDetail) {
-    lat = (mapRef.current.props.center as Array<number>)[0];
-    lng = (mapRef.current.props.center as Array<number>)[1];
-  }
   const parcelLayerFeature = useAppSelector(store => store.parcelLayerData?.parcelLayerFeature);
+  const zoom = useAppSelector(store => store.mapViewZoom) ?? zoomProp;
+
+  const [baseLayers, setBaseLayers] = React.useState<BaseLayer[]>([]);
+  const [showFilterBackdrop, setShowFilterBackdrop] = React.useState(true);
+  const [activeBasemap, setActiveBasemap] = React.useState<BaseLayer | null>(null);
+  const [bounds, setBounds] = React.useState<LatLngBounds>(defaultBounds);
+  const [layerPopup, setLayerPopup] = React.useState<LayerPopupInformation>();
+  const [geoFilter, setGeoFilter] = React.useState<IGeoSearchParams>({
+    ...defaultFilterValues,
+    includeAllProperties: keycloak.hasClaim(Claims.ADMIN_PROPERTIES),
+  } as any);
+  const [center, setCenter] = React.useState({ lat, lng });
+  const [infoOpen, setInfoOpen] = React.useState(false);
+  const [layersOpen, setLayersOpen] = React.useState(false);
+
   useActiveFeatureLayer({
     selectedProperty,
     layerPopup,
     mapRef,
     parcelLayerFeature,
     setLayerPopup,
   });
-  const [showFilterBackdrop, setShowFilterBackdrop] = useState(true);
 
-  const lastZoom = useAppSelector(store => store.mapViewZoom) ?? zoomProp;
-  const [zoom, setZoom] = useState(lastZoom);
-  useEffect(() => {
-    if (lastZoom === DEFAULT_MAP_ZOOM) {
-      dispatch(setMapViewZoom(smallScreen ? 4.9 : 5.5));
-    } else if (lastZoom !== zoom && zoom !== DEFAULT_MAP_ZOOM) {
-      dispatch(setMapViewZoom(zoom));
+  React.useEffect(() => {
+    // fetch GIS base layers configuration from /public folder
+    axios.get('/basemaps.json')?.then(result => {
```

Comment:
From what I've learned, making axios calls within components is generally considered malpractice.
Would you be able to abstract this functionality to a function in another file?

Reply:
I entirely agree with you, but I've no time to refactor.  This regrettably was code from the dev who originally implemented.

Reply:
No worries, totally makes sense to get the most functionality out of your time.

I'll make sure it gets done before merging in.